k8s/hyperconverged/tutorial-configure-openebs-aks.md

```diff
@@ -0,0 +1,264 @@
+
+3aab0f9a48e2    k8s-gcrio.azureedge.net/hyperkube-amd64:v1.8.7   "/hyperkube kubele..."   48 minutes ago      Up 48 minutes                           eager_einstein
+```
+
+Run the following commands to install and configure iSCSI in each node.
+
+```
+azureuser@aks-nodepool1-46849391-1:~$ sudo docker exec -it 3aab0f9a48e2 bash
+# apt-get update
+# apt install -y open-iscsi
+# exit
+```
+Check the status of iSCSI service by running the following command,
+```
+azureuser@aks-nodepool1-46849391-1:~$ service open-iscsi status
+● open-iscsi.service - Login to default iSCSI targets
+   Loaded: loaded (/lib/systemd/system/open-iscsi.service; enabled; vendor preset: enabled)
+   Active: active (exited) since Mon 2018-03-19 11:27:01 UTC; 21h ago
+     Docs: man:iscsiadm(8)
+           man:iscsid(8)
+ Main PID: 1497 (code=exited, status=0/SUCCESS)
+    Tasks: 0
+   Memory: 0B
+      CPU: 0
+   CGroup: /system.slice/open-iscsi.service
+
+Mar 19 11:27:03 aks-nodepool1-46849391-1 iscsiadm[1474]: iscsiadm: No records found
+Mar 19 11:27:01 aks-nodepool1-46849391-1 systemd[1]: Starting Login to default iSCSI targets...
+Mar 19 11:27:01 aks-nodepool1-46849391-1 systemd[1]: Started Login to default iSCSI targets.
+
+azureuser@aks-nodepool1-46849391-1:~$ exit
+devops@Azure:~$
+```
+
+**Note** : Install and configure iSCSI in all the nodes of your Kubernetes cluster by following the above procedure.
+
+## Setting up OpenEBS
+
+Download the latest OpenEBS repository using the following commands.
+
+```
+devops@Azure:~$ git clone https://github.com/openebs/openebs.git
+Cloning into 'openebs'...
+remote: Counting objects: 8081, done.
+remote: Compressing objects: 100% (90/90), done.
+remote: Total 8081 (delta 54), reused 42 (delta 16), pack-reused 7975
+Receiving objects: 100% (8081/8081), 8.67 MiB | 64.00 KiB/s, done.
+Resolving deltas: 100% (4587/4587), done.
+Checking connectivity... done.
+
+```
+
+To install OpenEBS components in Kubernetes cluster, run the following commands.
+
+```
+devops@Azure:~$ cd openebs/k8s
+
+devops@Azure:~/openebs/k8s$ kubectl apply -f openebs-operator.yaml
+serviceaccount "openebs-maya-operator" created
+clusterrole "openebs-maya-operator" created
+clusterrolebinding "openebs-maya-operator" created
+deployment "maya-apiserver" created
+service "maya-apiserver-service" created
+deployment "openebs-provisioner" created
+customresourcedefinition "storagepoolclaims.openebs.io" created
+customresourcedefinition "storagepools.openebs.io" created
+storageclass "openebs-standard" created
+customresourcedefinition "volumepolicies.openebs.io" created
+```
+OpenEBS provides a set of yaml files that illustrate the usage of its storage classes. This can be modified according to your requirements by editing openebs-storageclasses.yaml file. You can set replica count and the storage capacity of volumes for the respective storage classes according to your requirement in openebs-storageclasses.yaml as follows:
```

Comment:
You can modify the openebs-storageclasses.yaml file as per your requirement.
requirement in the openebs-storageclasses.yaml file as follows.